internal/clients/builder.go

```diff
@@ -68,9 +68,13 @@ func Build(ctx context.Context, builder ClientBuilder) (*Client, error) {
 		return nil, fmt.Errorf("unable to build authorizer for Storage API: %+v", err)
 	}
 
-	attestationAuth, err := auth.NewAuthorizerFromCredentials(ctx, *builder.AuthConfig, builder.AuthConfig.Environment.Attestation)
-	if err != nil {
-		return nil, fmt.Errorf("unable to build authorizer for Attestation API: %+v", err)
+	if _, ok := builder.AuthConfig.Environment.Attestation.Endpoint(); ok {
+		attestationAuth, err = auth.NewAuthorizerFromCredentials(ctx, *builder.AuthConfig, builder.AuthConfig.Environment.Attestation)
+		if err != nil {
+			return nil, fmt.Errorf("unable to build authorizer for Attestation API: %+v", err)
+		}
+	} else {
+		log.Printf("[DEBUG] Skipping building the Attestation Authorizer since this is not supported in the current Azure Environment")
```

Comment:
Should we make this debug message more specific and informative by mentioning which azure environment is not supported?

```suggestion
		log.Printf("[DEBUG] Skipping building the Attestation Authorizer since the Attestation Authorizer is not supported in the %q environment", builder.AuthConfig.Environment.Attestation)
```